File: ke17/ruby/main.rb
require_relative 'simple_ke17_lib'

seed = (ARGV[0] || srand()).to_i
srand(seed)

node = SimpleNode.new([3, 2, -1])

N = 1000000
checks = [
  [1,1,0],
  [1,-1,0],
  [0,1,0]
]
puts "--- # ruby(seed: #{seed})"
checks.each do |check|
  acc = 0
  N.times do
    acc += node.input(check)
  end
  print '- ', acc / N.to_f, "\n"
end

File: ke17/python/main.py
from sys import argv
import time
import random
import simple_ke17_lib


if len(argv) >= 2:
    seed = argv[1]
else:
    seed = time.time()
random.seed(seed)

node = simple_ke17_lib.SimpleNode([3, 2, -1])

N = 1000000
checks = [
    [1,1,0],
    [1,-1,0],
    [0,1,0]
]
print("--- # python(seed: %s)" % seed)
for check in checks:
    acc = 0
    for i in range(N):
        acc += node.input(check)
    print("- %s" % (acc / N))
